### ISSUE TITLE
Signal handling seems to not work as expected

### ISSUE THREAD

**Issue by dm3ch** [helmfile/helmfile#746]
### Operating system

MacOS 13.2.1

### Helmfile Version

 ▓▓▓ helmfile    Version            v0.151.0   Git Commit         "brew"   Build Date         17 Feb 23 08:09 EET (3 weeks ago)   Commit Date        17 Feb 23 08:09 EET (3 weeks ago)   Dirty Build        no   Go version         1.19.6   Compiler           gc   Platform           darwin/arm64

### Helm Version

version.BuildInfo{Version:"v3.11.2", GitCommit:"912ebc1cd10d38d340f048efaf0abda047c3468e", GitTreeState:"clean", GoVersion:"go1.20.2"}

### Bug description

When I'm trying to interrupt helmfile apply by `Ctrl+C` from console all works as expected, release is marked as failed and I'm getting `Canceled` messages in stderr.

But when I'm trying to interrupt helmfile by sending signals using kill (`kill -2 <pid>`, `kill -15 <pid>`) helmfiles just ignores them. So I find that only `kill -9 <pid>` is working, but of course -9 just kills process without marking releases as failed.

This errors seems to be not only limited to MacOS, cause workflow canceling in my Github Actions workflow is also not handled properly.

### Example helmfile.yaml

Doesn't matter

### Error message you've seen (if any)

No

### Steps to reproduce

Start long running helmfile apply, send SIGINT/SIGTERM using the kill from another terminal

### Working Helmfile Version

No

**Comment by dm3ch** [comment]
Related issues in the past:
https://github.com/helmfile/helmfile/issues/416

@yxxhero @mumoshu Mentioning you as contibutors/maintainers who were participating in previous issue.

**Comment by yxxhero** [comment]
@dm3ch Thanks for your feedback. and what is the expected behavior?

**Comment by dm3ch** [comment]
Expected that the behaviour of sending SIGINT via kill should be the same as when you are just pressing Ctrl+C in terminal running helmfile

**Comment by dm3ch** [comment]
@yxxhero I would like to make it. But unfortunately it seems that my knowledge is not enough.

As far as I understand the signal handling code (https://github.com/helmfile/helmfile/blob/main/main.go#L29-L40) should live in lambda function (https://github.com/helmfile/helmfile/blob/main/main.go#L20-L22) to be able to be handled immediately.

In my experiment moving mentioned logic to lambda and adding debug print I ensured, that signal is handled immediately (even when it's send by kill).

Unfortunately I don't understand how CleanWaitGroup supposed to work. So my change just hangs up on CleanWaitGroup and waits until successful end of apply (not propagates a signal to child processes)

**Comment by dm3ch** [comment]
I tried to just drop CleanWaitGroup.Wait() and all works as I expect, except cleaning temporary files.

**Comment by dm3ch** [comment]
I have done some PoC with rough context propagation and sending sigint to helm in case of context cancelation.
It seems to work as expected, but code is very hacky, unfortunately, I'm not familiar enough with both helmfile code base and golang.

I would be glad if someone review this and make suggests/comments.

By the way, I still don't understand why the current master handles ctrl+c (which as far as I know sends SIGINT) and kill -2 <pid> (which also sends SIGINT) different

**Comment by yxxhero** [comment]
Pelease fix DCO issue.

**Comment by dm3ch** [comment]
> Pelease fix DCO issue.

Done, but there's still multiple things to be done:
- Fix tests
- Get rid of code duplication (and some of context.Background()) that were done to simplify PoC

**Comment by xiaomudk** [comment]
@dm3ch Hi, will you continue to finish it? if not please tell me

**Comment by mumoshu** [comment]
Sorry for the delay @dm3ch! I finally had some time to test your fix. I used a modified version of https://github.com/steved/helmfile-wait-interrupt contributed by @steved along with the prev issue #416. Your fix worked perfectly!

Let me approve and merge the PR soon.

For reference, I've published the modified repro setup at https://github.com/mumoshu/helmfile-wait-interrupt/commit/3f9534ffe8b489d3dece9937e0d6b95703fbeb0a
